docs/guide/src/en/contributing/walkthrough_readme.md

```diff
@@ -63,64 +63,74 @@ pub struct VirtualDom {
 ```
 
 > What is a [slab](https://docs.rs/slab/latest/slab/)?
+>
 > A slab acts like a hashmap with integer keys if you don't care about the value of the keys. It is internally backed by a dense vector which makes it more efficient than a hashmap. When you insert a value into a slab, it returns an integer key that you can use to retrieve the value later.
 
 > How does Dioxus use slabs?
-> Dioxus uses "synchronized slabs" to communicate between the renderer and the VDOM. When an node is created in the Virtual Dom, a ElementId is passed along with the mutation to the renderer to identify the node. These ids are used by the Virtual Dom to reference that nodes in future mutations like setting an attribute on a node or removing a node.
-> When the renderer sends an event to the Virtual Dom, it sends the ElementId of the node that the event was triggered on. The Virtual Dom uses this id to find the node in the slab and then run the necessary event handlers.
+>
+> Dioxus uses "synchronized slabs" to communicate between the renderer and the VDOM. When a node is created in the Virtual DOM, an (elementId, mutation) pair is passed to the renderer to identify that node, which the renderer will then render in actual DOM. These ids are also used by the Virtual Dom to reference that node in future mutations, like setting an attribute on a node or removing a node. When the renderer sends an event to the Virtual Dom, it sends the ElementId of the node that the event was triggered on. The Virtual DOM uses this id to find that node in the slab and then run the necessary event handlers (Question - what are "necessary event handlers?").
 
-The virtual dom is a tree of scopes. A new scope is created for every component when it is first rendered and recycled when the component is unmounted.
+The virtual DOM is a tree of scopes. A new `Scope` is created for every component when it is first rendered and recycled when the component is unmounted (JSNote: unmounted and recycled are terms I don't fully understand in this context. Best guess: Unmounted = element a scope is for is deleted from the virtual dom? Recycled = a scope having its slab dealocated?)
```

Comment:
> recycled when the component is unmounted

Unmounted and recycled are terms I don't fully understand in this context. Best guess: Unmounted = element a scope is for is deleted from the virtual dom? Recycled = a scope having its slab dealocated?